File: app/src/main/java/com/yubico/yubikitold/application/oath/OathType.java
package com.yubico.yubikitold.application.oath;

public enum OathType {

    HOTP((byte) 0x10),
    TOTP((byte) 0x20);

    public final byte value;

    OathType(byte value) {
        this.value = value;
    }

    public static OathType fromValue(byte value) {
        for (OathType type : OathType.values()) {
            if (type.value == value) {
                return type;
            }
        }
        throw new IllegalArgumentException("Not a valid OathType");
    }
}

File: app/src/main/kotlin/com/yubico/yubioath/protocol/OathType.kt
package com.yubico.yubioath.protocol

sealed class OathType(val byteVal: Byte) {
    override fun toString() = javaClass.simpleName.toLowerCase()

    object HOTP : OathType(0x10)
    object TOTP : OathType(0x20)

    companion object {
        fun fromValue(byteVal: Byte) = when (byteVal) {
            0x10.toByte() -> HOTP
            0x20.toByte() -> TOTP
            else -> throw IllegalArgumentException()
        }
    }
}